Jenkinsfile

```diff
@@ -0,0 +1,409 @@
+  }
+  if (env.WDM_SERVICE_URL != null) {
+    dockerEnv+="WDM_SERVICE_URL=${env.WDM_SERVICE_URL}\n"
+  }
+  if (env.WORKSPACE != null) {
+    dockerEnv+="WORKSPACE=${env.WORKSPACE}\n"
+  }
+  writeFile file: DOCKER_ENV_FILE, text: dockerEnv
+}
+
+def generateSecretsFile = { ->
+  withCredentials([
+    string(credentialsId: '9f44ab21-7e83-480d-8fb3-e6495bf7e9f3', variable: 'CISCOSPARK_CLIENT_SECRET'),
+    string(credentialsId: 'CISCOSPARK_APPID_SECRET', variable: 'CISCOSPARK_APPID_SECRET'),
+    usernamePassword(credentialsId: 'SAUCE_LABS_VALIDATED_MERGE_CREDENTIALS', passwordVariable: 'SAUCE_ACCESS_KEY', usernameVariable: 'SAUCE_USERNAME'),
+    string(credentialsId: 'ddfd04fb-e00a-4df0-9250-9a7cb37bce0e', variable: 'COMMON_IDENTITY_CLIENT_SECRET')
+  ]) {
+    def secrets = ""
+    secrets += "COMMON_IDENTITY_CLIENT_SECRET=${COMMON_IDENTITY_CLIENT_SECRET}\n"
+    secrets += "CISCOSPARK_APPID_SECRET=${CISCOSPARK_APPID_SECRET}\n"
+    secrets += "CISCOSPARK_CLIENT_SECRET=${CISCOSPARK_CLIENT_SECRET}\n"
+    secrets += "SAUCE_USERNAME=${SAUCE_USERNAME}\n"
+    secrets += "SAUCE_ACCESS_KEY=${SAUCE_ACCESS_KEY}\n"
+    writeFile file: ENV_FILE, text: secrets
+  }
+}
+
+ansiColor('xterm') {
+  timestamps {
+    timeout(90) {
+
+      node("SPARK_JS_SDK_VALIDATING") {
+        try {
+          env.CONCURRENCY = 4
+          env.NPM_CONFIG_REGISTRY = "http://engci-maven-master.cisco.com/artifactory/api/npm/webex-npm-group"
+          env.ENABLE_VERBOSE_NETWORK_LOGGING = true
+          env.SDK_ROOT_DIR=pwd
+
+          if (JOB_NAME.toLowerCase().contains('validated-merge')) {
+            IS_VALIDATED_MERGE_BUILD = true
+          }
+
+          if (IS_VALIDATED_MERGE_BUILD) {
+            env.COVERAGE = true
+          }
+          else {
+            env.PIPELINE = true
+            env.SKIP_FLAKY_TESTS = true
+          }
+
+          DOCKER_IMAGE_NAME = "${JOB_NAME}-${BUILD_NUMBER}-builder"
+          def image
+
+          DOCKER_ENV_FILE = "${env.WORKSPACE}/docker-env"
+          ENV_FILE = "${env.WORKSPACE}/.env"
+
+          DOCKER_RUN_OPTS = ''
+          DOCKER_RUN_OPTS = "${DOCKER_RUN_OPTS} --env-file=${DOCKER_ENV_FILE}"
+          DOCKER_RUN_OPTS = "${DOCKER_RUN_OPTS} --env-file=${ENV_FILE}"
+          DOCKER_RUN_OPTS = "${DOCKER_RUN_OPTS} -e NPM_CONFIG_CACHE=${env.WORKSPACE}/.npm"
+          DOCKER_RUN_OPTS = "${DOCKER_RUN_OPTS} --volumes-from=\$(hostname)"
+          DOCKER_RUN_OPTS = "${DOCKER_RUN_OPTS} --user=\$(id -u):\$(id -g)"
+          // DOCKER_RUN_OPTS has some values in it that we want to evaluate on
+          // the node, but image.inside doesn't do subshell execution. We'll use
+          // echo to evaluate once on the node and store the values.
+          DOCKER_RUN_OPTS = sh script: "echo -n ${DOCKER_RUN_OPTS}", returnStdout: true
+          // image.inside uses the -d flag, so we can only use --rm for
+          // bash-started containers
+          env.DOCKER_RUN_OPTS = "${DOCKER_RUN_OPTS} --rm ${DOCKER_IMAGE_NAME}"
```

Comment:
Got ya... I was thinking it'd be easier to provide all the settings in a dockercompose file so that all you would need to do is docker-compose up and it'll bring everything online. But makes sense right now to change as little as possible.